commands/delete.py

```diff
@@ -5,40 +5,16 @@
 
 
 class FmDeleteCommand(AppCommand):
-    def delete(self, index):
-        if index == 0:
-            for path in self.paths:
-                view = self.window.find_open_file(path)
-                if view is not None:
-                    close_view(view)
-                try:
-                    send2trash(path)
-                except OSError as e:
-                    sublime.error_message("Unable to send to trash: {}".format(e))
-                    raise OSError("Unable to send {0!r} to trash: {1}".format(path, e))
-        if index > 1:
-            view = self.window.find_open_file(self.paths[index - 2])
-            if view is not None:
-                close_view(view)
-
-            # We substract two, because 0, 1 are populated by Confirm, Cancel
-            self.paths.remove(self.paths[index - 2])
-
-            if self.paths:
-                refresh_sidebar(self.settings, self.window)
-                self.run(self.paths)
-
-        refresh_sidebar(self.settings, self.window)
-
     def run(self, paths=None, *args, **kwargs):
         self.settings = get_settings()
         self.window = get_window()
         self.view = get_view()
 
         self.paths = paths or [self.view.file_name()]
-        if get_settings().get("ask_for_confirmation_on_delete") is not False:
-            nitems = "{0} ".format(len(self.paths)) if len(self.paths) > 1 else ""
-            extras = "s" if len(self.paths) > 1 else ""
+        if get_settings().get("ask_for_confirmation_on_delete"):
+            num_paths = len(self.paths)
+            nitems = "{0} ".format(num_paths) if num_paths > 1 else ""
```

Comment:
> Avoids calling len(self.paths) multiple times.

In python, I think the lengths of lists/strings are stored, unlike in C where they use a null byte at the end. So, [`len` is O(1)](https://wiki.python.org/moin/TimeComplexity). Hence, I think it's actually cleaner to make multiple call to `len` here.

Reply:
Agreed! Also, fun to know that python stores the length and is a O(1) operation. Very cheap!